git_rewrite_merge_moveit.py

```diff
@@ -0,0 +1,143 @@
+        shutil.rmtree(moveit_experimental_dir)
+
+    call(['git', 'clone', 'http://github.com/ros-planning/moveit_experimental.git', moveit_experimental_dir])
+    os.chdir(moveit_experimental_dir)
+    for branch in branches_to_merge:
+        call(['git', 'checkout', '-b', branch])
+    os.chdir('..')
+    repos_to_merge['moveit_experimental'] = 'file://{0}'.format(os.path.abspath(moveit_experimental_dir))
+
+    if os.path.exists(repo_dst):
+        print("Warning! '{0}' exists, it will be removed in 5 seconds..."
+              .format(repo_dst), file=sys.stderr)
+        time.sleep(5)
+        shutil.rmtree(repo_dst)
+
+    print("\n==> Creating git repository '{0}' to merge repositories into."
+          .format(repo_dst))
+    os.makedirs(repo_dst)
+    os.chdir(repo_dst)
+    call(['git', 'init', '.'])
+    call(['git', 'commit', '-m', 'initial commit before merging branches', '--allow-empty'])
+
+    # For each branch to be merged, setup a local branch to merge into.
+    for branch, distro in branches_to_merge.items():
+        print("\n==> Preparing branch '{0}'".format(branch))
+        call(['git', 'checkout', '-b', branch, 'master'])
+
+        print("\n==> Merging imported branches for branch '{0}'".format(branch))
+        for repo, addr in repos_to_merge.items():
+            print("==> Processing '{0}' branch from the '{1}' repository".format(branch, repo))
+            temp_branch = 'temp' + '/' + repo + '/' + branch
+            call(['git', 'checkout', '-b', temp_branch, branch])
+            call(['git', 'pull', addr, branch, '--allow-unrelated-histories', '--no-edit'])
+            call(['git', 'filter-branch', '-f',
+                  '--tree-filter',
+                  'mkdir -p {0}; test "$(ls -A | grep -v {0})" && mv $(ls -A | grep -v {0}) {0} || true'
+                  .format(repo), 'HEAD'])
+            print("==> Merging '{0}' branch from the '{1}' repository".format(branch, repo))
+            call(['git', 'checkout', branch])
+            call(['git', 'merge', temp_branch, '--allow-unrelated-histories', '--no-edit'])
+            call(['git', 'branch', '-d', temp_branch])
+            trashed_files = False
+            if os.path.isfile(os.path.join(repo, '.gitignore')):
+                trashed_files = True
+                call(['git', 'rm', os.path.join(repo, '.gitignore')])
+            if os.path.isfile(os.path.join(repo, '.travis.yml')):
+                trashed_files = True
+                call(['git', 'rm', os.path.join(repo, '.travis.yml')])
+            if trashed_files:
+                call(['git', 'commit', '-m', 'removing vestigial files after merging'])
+
+        print("\n==> Adding new README.md and .gitignore for branch '{0}'".format(branch))
+        template_file(
+            os.path.join(template_dir, 'README.md.template'),
+            os.path.join('README.md'),
+            {
+                '__DISTRO__': distro,
+                '__DISTRO_CAP__': distro.capitalize(),
```

Comment:
This didn't seem to work [here](https://github.com/wjwwood/moveit/commit/a517459271f2255df42685a5eed0a350bb3acc13#diff-04c6e90faac2675aa89e2176d2eec7d8R43)